pkg/lso/localvolumediscovery_test.go

```diff
@@ -0,0 +1,391 @@
+		assert.Equal(t, testCase.expectedStatus, exist)
+	}
+}
+
+func TestLocalVolumeDiscoveryGet(t *testing.T) {
+	testCases := []struct {
+		testLocalVolumeDiscovery *LocalVolumeDiscoveryBuilder
+		expectedError            error
+	}{
+		{
+			testLocalVolumeDiscovery: buildValidLVDObjectBuilder(buildLVDClientWithDummyObject()),
+			expectedError:            nil,
+		},
+		{
+			testLocalVolumeDiscovery: buildInValidLVDObjectBuilder(buildLVDClientWithDummyObject()),
+			expectedError:            fmt.Errorf("localVolumeDiscovery 'name' cannot be empty"),
+		},
+		{
+			testLocalVolumeDiscovery: buildValidLVDObjectBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError: fmt.Errorf("localvolumediscoveries.local.storage.openshift.io " +
+				"\"auto-discover-devices\" not found"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		localVolumeDiscoveryObj, err := testCase.testLocalVolumeDiscovery.Get()
+
+		if testCase.expectedError == nil {
+			assert.Equal(t, localVolumeDiscoveryObj.Name, testCase.testLocalVolumeDiscovery.Definition.Name)
+			assert.Equal(t, localVolumeDiscoveryObj.Namespace, testCase.testLocalVolumeDiscovery.Definition.Namespace)
+			assert.Nil(t, err)
+		} else {
+			assert.Equal(t, testCase.expectedError.Error(), err.Error())
+		}
+	}
+}
+
+func TestLocalVolumeDiscoveryCreate(t *testing.T) {
+	testCases := []struct {
+		testLocalVolumeDiscovery *LocalVolumeDiscoveryBuilder
+		expectedError            error
+	}{
+		{
+			testLocalVolumeDiscovery: buildValidLVDObjectBuilder(buildLVDClientWithDummyObject()),
+			expectedError:            nil,
+		},
+		{
+			testLocalVolumeDiscovery: buildInValidLVDObjectBuilder(buildLVDClientWithDummyObject()),
+			expectedError:            fmt.Errorf("localVolumeDiscovery 'name' cannot be empty"),
+		},
+		{
+			testLocalVolumeDiscovery: buildValidLVDObjectBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError:            nil,
+		},
+	}
+
+	for _, testCase := range testCases {
+		testLocalVolumeSetBuilder, err := testCase.testLocalVolumeDiscovery.Create()
+		assert.Equal(t, testCase.expectedError, err)
+
+		if testCase.expectedError == nil {
+			assert.Equal(t, testLocalVolumeSetBuilder.Definition.Name, testLocalVolumeSetBuilder.Object.Name)
+			assert.Equal(t, testLocalVolumeSetBuilder.Definition.Namespace, testLocalVolumeSetBuilder.Object.Namespace)
+			assert.Nil(t, err)
+		}
+	}
+}
+
+func TestLocalVolumeDiscoveryDelete(t *testing.T) {
+	testCases := []struct {
+		testLocalVolumeDiscovery *LocalVolumeDiscoveryBuilder
+		expectedError            error
+	}{
+		{
+			testLocalVolumeDiscovery: buildValidLVDObjectBuilder(buildLVDClientWithDummyObject()),
+			expectedError:            nil,
+		},
+		{
+			testLocalVolumeDiscovery: buildValidLVDObjectBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError:            fmt.Errorf("localVolumeDiscovery cannot be deleted because it does not exist"),
```

Comment:
Please fix Delete function. We don't return error if Object doesn't exist during Delete call

Reply:
Please fix Delete function

Reply:
yes, sorry, my laptop crashed when I worked on this change. I was sure it is in already